coderd/rbac/scopes.go

```diff
@@ -143,18 +144,63 @@ func AllowListAll() AllowListElement {
 	return AllowListElement{ID: policy.WildcardSymbol, Type: policy.WildcardSymbol}
 }
 
+// String encodes the allow list element into the canonical database representation
+// "type:id". This avoids fragile manual concatenations scattered across the codebase.
+func (e AllowListElement) String() string {
+	return e.Type + ":" + e.ID
+}
+
 func (s Scope) Expand() (Scope, error) {
 	return s, nil
 }
 
 func (s Scope) Name() RoleIdentifier {
-	return s.Role.Identifier
+	return s.Identifier
 }
 
 func ExpandScope(scope ScopeName) (Scope, error) {
-	role, ok := builtinScopes[scope]
-	if !ok {
-		return Scope{}, xerrors.Errorf("no scope named %q", scope)
+	if role, ok := builtinScopes[scope]; ok {
+		return role, nil
+	}
+	if res, act, ok := parseLowLevelScope(scope); ok {
+		return expandLowLevel(res, act), nil
+	}
+	return Scope{}, xerrors.Errorf("no scope named %q", scope)
+}
+
+// parseLowLevelScope parses a low-level scope name formatted as
+// "<resource>:<action>" and validates it against RBACPermissions.
+// Returns the resource and action if valid.
+func parseLowLevelScope(name ScopeName) (resource string, action policy.Action, ok bool) {
+	parts := strings.SplitN(string(name), ":", 2)
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		return "", "", false
+	}
```

Comment:
Should move this function to the rbac package and make the db type `Scan` call the same function?